core/services/assets/Registry.php

```diff
@@ -256,6 +283,57 @@ public function getData($key)
     }
 
 
+    /**
+     * Get the actual asset path for asset manifests.
+     * If there is no asset path found for the given $chunk_name, then the $chunk_name is returned.
+     * @param string $chunk_name
+     * @param string $asset_type
+     * @return string
+     * @throws InvalidFilePathException
+     * @since $VID:$
+     */
+    public function getAssetUrl($chunk_name, $asset_type)
+    {
+        if (empty($this->cached_manifests)) {
+            $this->registerManifests();
+        }
+        return isset($this->cached_manifests[$chunk_name][$asset_type])
+            ? $this->domain->distributionAssetsUrl() . $this->cached_manifests[$chunk_name][$asset_type]
+            : $chunk_name;
+    }
+
+
+    /**
+     * Register any asset manifests.
+     * @throws InvalidFilePathException
+     * @since $VID:$
+     */
+    private function registerManifests()
+    {
+        if (! empty($this->cached_manifests)) {
+            //already registered get out.  This usually means that a $chunk_name (argument for getAssetUrl call) either
+            //doesn't exist in any registered manifest files.  This could be because the $chunk_name is an actual file
+            //url and thus not registered in the manifest, or a plugin registered its manifest files too late.
+            return;
+        }
+        $registered_manifest_files = apply_filters(
+            'FHEE__EventEspresso_core_services_assets_Registry__registerManifests__registered_manifest_files',
+            $this->domain->distributionAssetsPath() . 'build-manifest.json'
+        );
+        foreach ($registered_manifest_files as $file) {
+            if (! file_exists($file)) {
+                throw new InvalidFilePathException($file);
+            }
+            /** recommended to avoid array_merge in loops */
+            $this->cached_manifests[] = json_decode(file_get_contents($file), true);
+        }
+        //PHP below 5.6
+        $this->cached_manifests = call_user_func_array('array_merge', $this->cached_manifests);
+        //We can use this once we support PHP5.6+
+        //$this->cached_manifests = array_merge(...$this->cached_manifests);
```

Comment:
I soooooo wish we could just use the spread syntax now 😖!  It would be so much more concise (and clearer imo).